### DISCUSSION TITLE
useTable without total being returned from the backend?

### DISCUSSION THREAD

**Discussion by vinclou** [refinedev/refine#2285] [Q&A]
It so happens the backend doesn't aggregate the total count of results and it's not being returned. In our react-query/app there's a custom react-query hook that prefetches the next page and sets the flag hasMore to false + lastPage. This creates an illusion of real pagination basically. We do not expect anyone to not use the search bar and actually paginate in search for a result.

- Is there are any patterns you can share to deal with this situation? 
- Or advice on how to go about it using Refine's API?

**Comment by omeraplak** [comment]
Hey @vinclou 👋,
Would it actually be helpful to be able to use `useInfiniteQuery` with `useTable`, `useSimpleList`, etc. ? I'm not sure how we can handle this, but I think we can get it on our backlog. Would this work for you?

For example, like Ant Design's "Load More" example,

https://ant.design/components/list/#components-list-demo-loadmore

**Reply by vinclou** [reply]
I'll dig into it; useInfiniteQuery is similar to useQuery when you set keepPreviousData: true; The reason we haven't necessarily went for it with our Ant tables back in the day. Is cause u need to unwrap pages; no one wanted to deal with refactoring Table components themselves, thus a small team. But essentially our custom pagination component is the classic Load More button, with an ability to jump to a page. I'll get back on this thread when I get more familiar with Refine's API. But I can see a couple of solutions here, just unfamiliar with the limits of the data hooks you provide. Honestly, though it's funny that libraries like AntD don't provide the simplest case of pagination(prev/next) out of the box.
